### PR TITLE
Logind session killing fix

### PR DESCRIPTION
**Pull request by keszybz**
KillUserProcesses=yes wasn't working.

### CODE REVIEW

**Comment by vcaputo** [comment]
lgtm, but I'm not familiar with this area of systemd.

File: src/login/logind-user.c [hunk 1]
```diff
@@ -541,7 +541,7 @@ static int user_remove_runtime_path(User *u) {
 
         r = rm_rf(u->runtime_path, 0);
         if (r < 0)
-                log_error_errno(r, "Failed to remove runtime directory %s: %m", u->runtime_path);
+                log_error_errno(r, "Failed to remove runtime directory %s (before unmounting): %m", u->runtime_path);
```
Comment (poettering) [inline src/login/logind-user.c:544]: maybe similarly check the message of the other rm_rf() call? that way things are easier to debug should users report issues with old vs. new logind versions

**Comment by keszybz** [comment]
Pfff, I updated the code, but forgot to push this. Updated with the suggested change.

So... how do we proceed with this? There are doubts (I have them too :]) if this works correctly in all circumstances. Maybe we should merge and ask people to test in various setups? Tibbs tested this, and reported some pulseaudio processes remaining after scope termination, but I think that this must be some different issue.

**Comment by jasontibbitts** [comment]
I'm back looking at this after a systemd update in Fedora 26 meant I had to re-patch the package.

I'm confident that any processes persisting after systemd terminates the session exist because they either trap the signal or somehow get restarted by some other process.  They do all die not long afterwards.  I suppose systemd is waiting for a bit and then sending SIGKILL, though I don't think the documentation says how long it waits.

In any case, with this patch applied I wasn't able to reproduce any processes surviving a session termination which weren't supposed to survive.  After the package updated and I was running without this patch, all of my previous issues returned.

**Pull request merged**